lib/foreman_openscap/helper.rb

```diff
@@ -10,8 +10,24 @@
 
 module ForemanOpenscap::Helper
   def self.get_asset(cname, policy_id)
-    asset = Host.find_by_name!(cname).get_asset
+    asset = find_host_by_name_or_uuid(cname).get_asset
     asset.policy_ids += [policy_id]
     asset
   end
+
+  private
+
+  def self.find_host_by_name_or_uuid(cname)
+    if defined?(Katello::System)
+      host = Host.includes(:content_host).where(:katello_systems => {:uuid => cname}).first
+      host ||= Host.find_by_name(cname)
+    else
+      host = Host.find_by_name(cname)
+    end
+    unless host
+      Rails.logger.error "Could not find Host with name: #{cname}"
```

Comment:
Oh fair enough, probably not necessary.  

If a host is reporting openscap, it should have a foreman host because it came via Puppet. It's Katello's problem if this situation happens, that there's no corresponding foreman host.